ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py

```diff
@@ -591,6 +592,52 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
             if any(wan_interface["connected_to_pathfinder"] for wan_interface in path_group["interfaces"])
         ]
 
+    def get_internet_exit_nat_profile(self, internet_exit_policy_type: Literal["zscaler", "direct"]) -> Tuple[dict | None, dict | None]:
+        if internet_exit_policy_type == "zscaler":
+            return "IE-ZSCALER-NAT"
+        else:
+            return "IE-DIRECT-NAT"
+
+    def get_internet_exit_nat(self, internet_exit_policy_type: Literal["zscaler", "direct"]) -> Tuple[dict | None, dict | None]:
+        if internet_exit_policy_type == "zscaler":
+            pool = {
+                "name": "PORT-ONLY-POOL",
+                "type": "port-only",
+                "ranges": [
+                    {
+                        "first_port": 1500,
+                        "last_port": 65535,
+                    }
+                ],
+            }
+
+            profile = {
+                "name": self.get_internet_exit_nat_profile(internet_exit_policy_type),
+                "source": {
+                    "dynamic": [
+                        {
+                            "access_list": "ALLOW-ALL",
+                            "pool_name": "PORT-ONLY-POOL",
+                            "nat_type": "pool",
+                        }
+                    ]
+                },
+            }
+            return pool, profile
+        elif internet_exit_policy_type == "direct":
+            profile = {
+                "name": self.get_internet_exit_nat_profile(internet_exit_policy_type),
+                "source": {
+                    "dynamic": [
+                        {
+                            "access_list": "ALLOW-ALL",
+                            "nat_type": "overload",
+                        }
+                    ]
+                },
+            }
+            return None, profile
+
```

Comment:
Since you only have `if` and `elif` I suggest you have a last resort to not get issues unpacking in the calling code.

```suggestion

        return None, None
```